engine/src/main/java/org/terasology/rendering/dag/nodes/CopyToVRFrameBuffersNode.java

```diff
@@ -0,0 +1,118 @@
+import org.terasology.rendering.world.WorldRenderer.RenderingStage;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+import static org.terasology.rendering.opengl.ScalingFactors.FULL_SCALE;
+
+public class CopyToVRFrameBuffersNode extends ConditionDependentNode {
+    private static final ResourceUrn LEFT_EYE_FBO = new ResourceUrn("engine:leftEye");
+    private static final ResourceUrn RIGHT_EYE_FBO = new ResourceUrn("engine:rightEye");
+    // TODO: make these configurable options
+
+    @In
+    private OpenVRProvider vrProvider;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private ScreenGrabber screenGrabber;
+
+    @In
+    private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
+
+    private RenderingConfig renderingConfig;
+    private Material finalPost;
+    private FBO leftEye;
+    private FBO rightEye;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        finalPost = worldRenderer.getMaterial("engine:prog.post");
+        requiresCondition(() -> renderingConfig.isVrSupport());
+        leftEye = requiresFBO(new FBOConfig(LEFT_EYE_FBO, FULL_SCALE,
+                FBO.Type.DEFAULT).useDepthBuffer(),displayResolutionDependentFBOs);
+        rightEye = requiresFBO(new FBOConfig(RIGHT_EYE_FBO,FULL_SCALE,
+                FBO.Type.DEFAULT).useDepthBuffer(),displayResolutionDependentFBOs);
+        if (this.vrProvider != null) {
+            logger.info("Left eye FBOID:" + Integer.toString(leftEye.fboId));
+            logger.info("Right eye FBOID:" + Integer.toString(leftEye.fboId));
+            logger.info("OpenVR init done.");
+            vrProvider.texType[0].handle = leftEye.colorBufferTextureId;
+            vrProvider.texType[0].eColorSpace = JOpenVRLibrary.EColorSpace.EColorSpace_ColorSpace_Gamma;
+            vrProvider.texType[0].eType = JOpenVRLibrary.EGraphicsAPIConvention.EGraphicsAPIConvention_API_OpenGL;
+            vrProvider.texType[0].write();
+            vrProvider.texType[1].handle = rightEye.colorBufferTextureId;
+            vrProvider.texType[1].eColorSpace = JOpenVRLibrary.EColorSpace.EColorSpace_ColorSpace_Gamma;
+            vrProvider.texType[1].eType = JOpenVRLibrary.EGraphicsAPIConvention.EGraphicsAPIConvention_API_OpenGL;
+            vrProvider.texType[1].write();
+        }
+    }
+
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/copyToVRFrameBuffers");
+        logger.info("Process");
+        finalPost.enable();
+        if (!renderingConfig.isVrSupport()) {
+            logger.warn("CopyToVRFrameBufferNode processed in non-VR mode.");
+        } else {
+            renderFinalStereoImage(worldRenderer.getCurrentRenderStage());
+        }
+        PerformanceMonitor.endActivity();
+    }
+
+    private void renderFinalStereoImage(RenderingStage renderingStage) {
```

Comment:
I'm on board with that, except for one detail. Do you really think that we need a split screen rendering in the default frame buffer? I personally think it would be better to just copy the left eye to the screen frame buffer. It's just as cheap (maybe cheaper), so at this point it's entirely a stylistic decision. It's a decent visual indication that VR rendering is in progress, allowing the user to be less confused if they start things up without the headset on, but for an outside observer it's easier to see what's going on in game if only a single frame buffer is copied to the screen.

The SDK also has an example of querying the distortion pattern and performing distortion on the image that is sent to the screen. Note that it's done after the frame is submitted to the compositor; it's entirely for visual effect on the rendering window. I'm personally not a fan, because, once again, there is some utility in somebody outside of VR being able to clearly observe what the player is seeing.

Anyway, I will get to this anywhere between the next couple hours to couple days.